stm32: Enhance error message fir bbram/rtc misconfiguration

### Discussed in https://github.com/zephyrproject-rtos/zephyr/discussions/55486

<div type='discussions-op-text'>

<sup>Originally posted by **lmikl** March  6, 2023</sup>
HI,

i would like used the bbram on a stm32f4 but i have a time an compilation error

I have configured an overlay for the board nucleo-f429zi :
 / {
	aliases {
		backup-register = &bbram;
	};
};

&bbram {
	status = "okay";
};

write this onfig in the prj.comf : CONFIG_BBRAM=y

and i have wrote this code to use it : 

static const struct device *backup_reg = NULL;

int main(void)
{
     backup_reg = DEVICE_DT_GET(DT_ALIAS(backup_register));

    if(backup_reg == NULL){
        LOG_INF("backup_reg device was not found");
    }
    else
        LOG_INF("backup_reg found");

    if (!device_is_ready(backup_reg)) {
		LOG_INF("backup_reg not ready");
    }
}

But i have all time this error : undefined reference to `__device_dts_ord_105'.

If someone can help me.
thanks in advance.</div>

--- a/dts/bindings/memory-controllers/st,stm32-bbram.yaml
+++ b/dts/bindings/memory-controllers/st,stm32-bbram.yaml
@@ -10,6 +10,9 @@ description: |
   by system reset or when the device wakes up from Standby mode. They are
   reset by a backup domain reset.
 
+  Note: Since STM32 BBRAM is in RTC backup domain, STM32 RTC node must be
+        enabled if STM32 BBRAM node is enabled.
+
 compatible: "st,stm32-bbram"
 
 include: base.yaml

